scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+            Name of ylabel
+        savename : str
+            Name of file to save as
+
+
+        """
+        self.data = {}
+        self.data['xs'] = xs
+        self.data['ys'] = ys
+        self.data['labs'] = labs
+        self.data['xlab'] = xlab
+        self.data['ylab'] = ylab
+        self.data['savename'] = savename
+        self.data['parasitic'] = self.parasitic_results
+        num_nucs = []
+        for i, x in enumerate(self.data['xs']):
+            num_nucs.append(np.shape(self.data['ys'][i])[2])
+
+        num_nucs = max(num_nucs)
+        for nuclide_i in range(num_nucs):
+            final_data_points = []
+            final_names = []
+            for i, x in enumerate(self.data['xs']):
+                nuclide = self.data_params['tracked_nucs'][nuclide_i]
+                try:
+                    y = np.mean(self.data['ys'][i][:, :, nuclide_i], axis=1)
+                except IndexError:
+                    continue
+                lab = f'{self.data["labs"][i]} {nuclide}'
+                self.data[f'spat_avg_y_method{i}_nuc{nuclide_i}'] = y
+                self.data[f'lab_method{i}_nuc{nuclide_i}'] = lab
+                final_data_points.append(y[-1])
+                final_names.append(lab)
+
+            for data_i, data in enumerate(final_data_points):
+                end_data = final_data_points[-1]
+                pcnt_diff = (data - end_data) / (end_data) * 100
+                name = final_names[data_i]
+                print(f'% diff {name} from most accurate: {pcnt_diff}%')
+        return
+
+    def _parasitic_abs(self, result_mat):
+        """
+        Calculates the amount of parasitic absorption for each nuclide
+            as atoms per cc as a function of time
+
+        Parameters
+        ----------
+        result_mat : :class:`np.ndarray`
+            Holds values over time, space, and nuclide (in that order)
+
+        Returns
+        -------
+        parasitic_abs_val : :class:`np.ndarray`
+            Parasitic absorption rates for nuclides over time in
+            atoms per cc
+        """
+        parasitic_abs_val = []
+        for nuclide in range(self.run_params['num_nuclides']):
+            paras_rate = []
+            for ti, t in enumerate(self.run_params['times']):
+                spat_avg_conc = np.mean(result_mat[ti, :, nuclide])
+                parasitic = spat_avg_conc * \
+                    self.data_params['loss_rates'][nuclide]
+                paras_rate.append(parasitic)
+            integral_form = integrate.cumtrapz(
+                paras_rate, self.run_params['times'])
+            integral_form = np.insert(integral_form, 0, 0.0)
+            parasitic_abs_val.append(integral_form)
+        return parasitic_abs_val
+
+    def ode_pde_compare(self):
+        """
+        Run with current parameters for both ODE and PDE solver methods.
+
+        Returns
+        -------
+        data : dict
+            key : str
+                Name of variable
```

Comment:
Please be a little more explicit. Ideally, explain what both the key and the values should be in a dict data object.